src/cloudai/parser.py

```diff
@@ -71,24 +81,63 @@ def parse(
         except TestConfigParsingError:
             exit(1)  # exit right away to keep error message readable for users
 
-        logging.debug(f"Parsed {len(tests)} tests: {[t.name for t in tests]}")
-        test_mapping = {t.name: t for t in tests}
+        try:
+            plugin_tests = (
+                self.parse_tests(list(PLUGIN_TEST_ROOT.glob("*.toml")), system) if PLUGIN_TEST_ROOT.exists() else []
+            )
+        except TestConfigParsingError:
+            exit(1)  # exit right away to keep error message readable for users
 
-        filtered_tests = tests
-        test_scenario: Optional[TestScenario] = None
-        if test_scenario_path:
+        if not test_scenario_path:
+            all_tests = list({test.name: test for test in tests + plugin_tests}.values())
+            return system, all_tests, None
+
+        test_mapping = {t.name: t for t in tests}
+        plugin_test_scenario_mapping = {}
+        if PLUGIN_ROOT.exists() and list(PLUGIN_ROOT.glob("*.toml")):
```

Comment:
Let's log.debug the existence of these directories and files, could be useful in future.